sw/device/silicon_creator/lib/base/rom_mmio.c

```diff
@@ -0,0 +1,19 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/base/rom_mmio.h"
+
+uint32_t rom_mmio_read8(uint32_t addr) { return ((volatile uint8_t *)addr)[0]; }
+
+void rom_mmio_write8(uint32_t addr, uint32_t value) {
+  ((volatile uint8_t *)addr)[0] = value;
+}
+
+uint32_t rom_mmio_read32(uint32_t addr) {
+  return ((volatile uint32_t *)addr)[0];
+}
+
+void rom_mmio_write32(uint32_t addr, uint32_t value) {
+  ((volatile uint32_t *)addr)[0] = value;
+}
```

Comment:
Should these be inline functions in the header file?

I assume they're implemented like this to enable tests to link with similarly-named symbols from the mock library.  If we can rely on LTO to inline them, then they probably can stay like this.

Reply:
+1 we need to be careful about not introducing glitchable function calls.

Reply:
Should I make these functions inline and add support for `-DMOCK_MMIO` of `-DMOCK_<MODULE_NAME>`?

Reply:
I think making them inline would be more explicit about what we want and expect the compiler to do.

Reply:
Done